src/probe_modules/module_dns.c

```diff
@@ -0,0 +1,1039 @@
+            // Mark byte consumed after we check for first iteration.
+
+            // Do we have enough data left (must have null byte too)?
+            if ((byte+1) > data_len) {
+                log_trace("dns", "_get_name_helper OUT. ERR. Not enough data for segment %hd"); 
+                return 0;
+            }   
+
+            // If we've consumed any bytes and are in a label, we're in a 
+            // label chain. We need to add a dot.
+            if (bytes_consumed > 0) { 
+
+                if (name_len < 1) {
+                    log_warn("dns", "Exceeded static name field allocation.");
+                    return 0;
+                }
+            
+                name[0] = '.';
+                name++;
+                name_len--;
+            }
+
+            // Now we've consumed a byte.
+            ++bytes_consumed;
+
+            // Did we run out of our arbitrary buffer?
+            if (byte > name_len) {
+                log_warn("dns", "Exceeded static name field allocation.");
+                return 0;
+            }
+
+            assert(data_len > 0);
+
+            memcpy(name, data, byte);
+    
+            name += byte;
+            name_len -= byte;
+    
+            data_len -= byte;
+            data += byte;
+            bytes_consumed += byte;
+ 
+            // Handled in the byte+1 check above.
+            assert(data_len > 0);
+        }
+    }
+
+    // We should never get here.
+    // For each byte we either have:
+    // -- a ptr, which terminates
+    // -- a null byte, which terminates
+    // -- a segment length which either terminates or ensures we keep looping
+
+    assert(0);
+    return 0;
+}
+
+// data: Where we are in the dns payload
+// payload: the entire udp payload
+char* get_name(const char* data, uint16_t data_len, const char* payload, 
+        uint16_t payload_len, uint16_t* bytes_consumed)
+{
+    log_trace("dns", "call to get_name, data_len: %d", data_len);
+
+    char* name = xmalloc(MAX_NAME_LENGTH);
+    
+    *bytes_consumed = get_name_helper(data, data_len, payload, 
+            payload_len, name, MAX_NAME_LENGTH - 1, 0);
+
+    if (*bytes_consumed == 0) {
+        free(name);
+        return NULL;
+    }
+
+    // Our memset ensured null byte.
+    assert(name[MAX_NAME_LENGTH - 1] == '\0');             
+
+    log_trace("dns", "return success from get_name, bytes_consumed: %d, string: %s", *bytes_consumed, name);
+
+    return name;
+
+}
+
+static bool process_response_question(char **data, uint16_t* data_len, const char* payload,
+        uint16_t payload_len, fieldset_t* list)
+{   
+    // Payload is the start of the DNS packet, including header
+    // data is handle to the start of this RR
+    // data_len is a pointer to the how much total data we have to work with.
+    // This is awful. I'm bad and should feel bad.
+    uint16_t bytes_consumed = 0;
+
+    char* question_name = get_name(*data, *data_len, payload, payload_len,  &bytes_consumed);
```

Comment:
This is over 80 characters. Can fix the 80 character limit for all lines that are not string literals?